### PR TITLE
feat: fetch assets from opensea

### PR DESCRIPTION
**Pull request by alaibe**
Opensea API integration to fetch collections and assets of collectible

### CODE REVIEW

File: services/wallet/opensea.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+		if len(assets) >= limit {
+			break
+		}
+	}
+	return assets, nil
+}
+
+func (o *OpenseaClient) doOpenseaRequest(url string) ([]byte, error) {
+	req, err := http.NewRequest("GET", url, nil)
```
Comment (richard-ramos) [inline services/wallet/opensea.go]:
```suggestion
	req, err := http.NewRequest(http.MethodGet, url, nil)
```

File: services/wallet/opensea.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	OwnedAssetCount int    `json:"owned_asset_count"`
+}
+
+type OpenseaClient struct {
+	client *http.Client
+}
+
+// new opensea client.
+func newOpenseaClient() *OpenseaClient {
+	client := &http.Client{}
```
Comment (richard-ramos) [inline services/wallet/opensea.go]:
```suggestion
	client := &http.Client{
		Timeout: time.Second * 5,
	}
```

**Review by richard-ramos — COMMENTED** [review]
Can we add a quick test unit? :)

File: services/wallet/opensea.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := o.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			log.Println(err.Error())
```
Comment (richard-ramos) [inline services/wallet/opensea.go]:
```suggestion
			log.Error("failed to close opensea request body", "err", err)
```

**Comment by alaibe** [comment]
@richard-ramos test added and suggestion applied
image

**Review by richard-ramos — APPROVED** [review]
:medal_military:

**Comment by iurimatias** [comment]
tested on https://github.com/status-im/status-desktop/pull/3240

**Pull request merged**